x-pack/filebeat/input/salesforce/input_manager_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package salesforce
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	v2 "github.com/elastic/beats/v7/filebeat/input/v2"
+	cursor "github.com/elastic/beats/v7/filebeat/input/v2/input-cursor"
+	"github.com/elastic/beats/v7/libbeat/statestore"
+	"github.com/elastic/beats/v7/libbeat/statestore/storetest"
+	conf "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/go-concert/unison"
+)
+
+// compile-time check if querier implements InputManager
+var _ v2.InputManager = InputManager{}
```

Comment:
Since InputManager is declared in `input_manager.go`, it should be moved to the same file and it would still have the same effect making sure that the code will fail to compile if `InputManager` type ever stops implementing the InputManager interface.